src/molecule/__main__.py

```diff
@@ -21,5 +21,17 @@
 
 from molecule.shell import main
 
+
+def mol_deprecation_notice():
+    print(
```

Comment:
We should probably use the [warnings](https://docs.python.org/3/library/warnings.html#module-warnings) module here since there is already support for deprecating things.